transport/grpc/src/main/java/io/a2a/transport/grpc/handler/GrpcHandler.java

```diff
@@ -312,17 +318,53 @@ private <V> ServerCallContext createCallContext(StreamObserver<V> responseObserv
-            // In Java: the proper equivalent would be ServerCall<ReqT,RespT> + Metadata from ServerInterceptor
-            // Current compromise: Store StreamObserver for basic functionality, but it lacks rich context
-            // 
-            // FUTURE ENHANCEMENT: Implement ServerInterceptor to capture ServerCall + Metadata,
-            // store in gRPC Context using Context.Key, then access via Context.current().get(key)
-            // This would provide proper equivalence to Python's ServicerContext
+            // Enhanced gRPC context access - equivalent to Python's grpc.aio.ServicerContext
+            // The A2AExtensionsInterceptor captures ServerCall + Metadata and stores them in gRPC Context
+            // This provides proper equivalence to Python's ServicerContext for metadata access
+            // Note: StreamObserver is still stored for response handling
             state.put("grpc_response_observer", responseObserver);
             
-            return new ServerCallContext(user, state);
+            // Add rich gRPC context information if available (set by interceptor)
+            // This provides equivalent functionality to Python's grpc.aio.ServicerContext
+            try {
+                Context currentContext = Context.current();
+                if (currentContext != null) {
+                    state.put("grpc_context", currentContext);
+                    
+                    // Add specific context information for easy access
+                    io.grpc.Metadata grpcMetadata = GrpcContextKeys.METADATA_KEY.get(currentContext);
+                    if (grpcMetadata != null) {
+                        state.put("grpc_metadata", grpcMetadata);
+                    }
+                    
+                    String methodName = GrpcContextKeys.METHOD_NAME_KEY.get(currentContext);
+                    if (methodName != null) {
+                        state.put("grpc_method_name", methodName);
+                    }
+                    
+                    String peerInfo = GrpcContextKeys.PEER_INFO_KEY.get(currentContext);
+                    if (peerInfo != null) {
+                        state.put("grpc_peer_info", peerInfo);
+                    }
+                }
+            } catch (Exception e) {
+                // Context not available - continue without it
+            }
+            
+            // Extract requested extensions from gRPC context (set by interceptor)
+            Set<String> requestedExtensions = new HashSet<>();
+            try {
+                // Try to get extensions from gRPC context (available when interceptor is used)
+                String extensionsHeader = getExtensionsFromContext();
+                if (extensionsHeader != null) {
+                    requestedExtensions = A2AExtensions.getRequestedExtensions(List.of(extensionsHeader));
+                }
+            } catch (Exception e) {
```

Comment:
Can an exception actually be caught here? It doesn't look like getExtensionsFromContext() or getRequestedExtensions() throw an exception. Let me know if I'm missing something.